Azure/communication-ui-library-ios
# [Feature] Implement message send status (Sending, delivered, failed)

## Purpose

* Implemented sending, delivered and failed status icon

## Does this introduce a breaking change?



## Pull Request Type
What kind of change does this Pull Request introduce?



## Pull request checklist

This PR has considered:

## How to Test


* Open a chat thread
* Send out a message without all other participants viewed it.
* Sent icon should appear on the last message
* Replace the following function
```
    func sendMessage(internalId: String,
                     content: String,
                     state: AppState,
                     dispatch: @escaping ActionDispatch) -> Task<Void, Never> {
        Task {
            do {
                guard let displayName = state.chatState.localUser?.displayName else {
                    return
                }
                let actualId = try await chatService.sendMessage(
                    content: content,
                    senderDisplayName: displayName)
                dispatch(.repositoryAction(.sendMessageSuccess(
                    internalId: internalId,
                    actualId: actualId)))
            } catch {
                dispatch(.repositoryAction(.sendMessageFailed(internalId: internalId, error: error)))
            }
        }
    }
```

with 
```
    func sendMessage(internalId: String,
                     content: String,
                     state: AppState,
                     dispatch: @escaping ActionDispatch) -> Task<Void, Never> {
        Task {
            do {
            } catch {
                dispatch(.repositoryAction(.sendMessageFailed(internalId: internalId, error: error)))
            }
        }
    }
```
to test sending status.

with 
```
    func sendMessage(internalId: String,
                     content: String,
                     state: AppState,
                     dispatch: @escaping ActionDispatch) -> Task<Void, Never> {
        Task {
                dispatch(.repositoryAction(.sendMessageFailed(internalId: internalId, error: NSError())))
        }
    }
```
to test failed status
                

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |
* Before
<img src="https://user-images.githubusercontent.com/107075081/202996019-c95e76ed-46c9-4482-9fac-fcd870ac75e3.png" width="200" height="400" />
* After
<img src="https://user-images.githubusercontent.com/107075081/202996266-af3e63b4-d184-4675-8ca1-38008d7344f7.png" width="200" height="400" /> <img src="https://user-images.githubusercontent.com/107075081/202996891-e0835522-f9e1-40f6-8e3e-61790edebb8f.png" width="200" height="400" /> <img src="https://user-images.githubusercontent.com/107075081/202996783-fb27d3db-8c62-479f-bce2-bc50e72881e0.png" width="200" height="400" />
